ext4/xattr.py

```diff
@@ -143,13 +156,22 @@ def size(self):
 
     @property
     def name_str(self):
-        if 0 > self.e_name_index or self.e_name_index > len(
+        name_index = self.e_name_index
+        if 0 > name_index or name_index >= len(
             ExtendedAttributeEntry.NAME_INDICES
         ):
-            raise ExtendedAttributeError(
+            warnings.warn(
```

Comment:
This should be behind an ignore flag on Volume, and raise the error if it is not set.